src/utils/flatten.js

```diff
@@ -55,7 +56,13 @@ const flatten = (
     /* Either execute or defer the function */
     if (typeof chunk === 'function') {
       if (executionContext) {
-        ruleSet.push(...flatten([chunk(executionContext)], executionContext))
+        const nextChunk = chunk(executionContext)
+        /* Throw if a React Element was given styles */
+        if (React.isValidElement(nextChunk)) {
+          const elementName = chunk.displayName || chunk.name
```

Comment:
Small nit; I’m not sure whether it’ll actually catch components here since they’re not elements so the displayName access is potentially dead code?

If I’m mistaken. We also have a utility for component names: https://github.com/bhargav-kitchen/styled-components/blob/aabb2bb0abc90b9c7a2d9822085be05d7505a07c/src/utils/getComponentName.js

Anyway, this might just be a thing for future PRs

Reply:
The babel react preset includes https://babeljs.io/docs/en/babel-plugin-transform-react-display-name which sets displayName on the classes